orderer/common/multichannel/registrar.go

```diff
@@ -370,3 +371,13 @@ func (r *Registrar) NewChannelConfig(envConfigUpdate *cb.Envelope) (channelconfi
 func (r *Registrar) CreateBundle(channelID string, config *cb.Config) (channelconfig.Resources, error) {
 	return channelconfig.NewBundle(channelID, config, r.bccsp)
 }
+
+func (r *Registrar) ListAllChannels() ([]channelparticipation.ChannelInfoShort, string) {
```

Comment:
I'm not sure the registrar should return a slice of this type. This creates an unnecessary dependency on this package. Perhaps define a type in a leaf package and then have the registrar return a slice of that type?

Reply:
I was thinking the same way and originally had the `ListAllChannels` return  `([]string, string)`.... But then, when I crafted the 

`ListChannel(channelID string) (*channelparticipation.ChannelInfoFull, error)` 

I realized these two packages are going to be linked anyway... so I gave up on that direction. 
The way it is now we don't have to copy strings from one struct to another...

Putting these types in a leaf package to break the cycle will help.